src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -196,6 +196,19 @@ public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsAsync(Diagnosti
             }
         }
 
+        private async Task<ImmutableArray<Diagnostic>> GetSourceGeneratorDiagnosticsAsync(Project project, CancellationToken cancellationToken)
```

Comment:
The try/catch in this method follows the pattern used in the prior method in this file that invokes into `InProcOrRemoteHostAnalyzerRunner`